bin/improver-spot-extract

```diff
@@ -109,6 +133,13 @@ def main():
         help="If provided, this JSON file can be used to modify the metadata "
         "of the returned netCDF file. Defaults to None.")
 
+    output_group = parser.add_argument_group("Suppress Verbose output")
+    output_group.add_argument(
+        "--suppress_warnings", default=False, action="store_true",
+        help="Suppress warning output. This option should only be used if "
+        "it is known that warnings will be generated but they are not "
+        "required.")
```

Comment:
What's the specific reason for this?

Reply:
This CLI can be used to replace the ver_stream_spot_preprocessing script in the suite (pretty much entirely). That script rattles through all the inputs provided to it, turning them all into spot-cubes regardless of their type, then it extracts the relevant percentile, or returns the untouched spot-extracted level-1 deterministic cube if that's what has been provided. This CLI does much the same, but it raises a warning if you provide a deterministic cube (e.g. UKVX level-1) and try to extract a percentile. To avoid filling the ver_stream_spot_preprocessing step logs with thousands of these known warnings (as we know we will provide level-1 data to it), I put in the suppress_warnings flag.

Reply:
That seems reasonable.  Is this something that can be documented here?  My concern is this looks very general and would be easy to misuse, but also to use as precedent for suppressing warnings in other CLIs where the behaviour is less justifiable (we should be sorting out the warnings at source).

Reply:
I've added a long comment next to the option in the CLI specifying its intended use. I think it is too much information for the help string, but I can put it there if you prefer.